edit/lint.js

```diff
@@ -0,0 +1,378 @@
+  const makeLink = (url, txt) => `<a target="_blank" href="${url}">${txt}</a>`;
+  const linter = prefs.get('editor.linter');
+  const url = linter === 'stylelint'
+    ? 'https://stylelint.io/user-guide/rules/'
+    // some CSSLint rules do not have a url
+    : 'https://github.com/CSSLint/csslint/issues/535';
+  const rules = [];
+  let template;
+  let list = '<ul class="rules">';
+  let header = '';
+  if (linter === 'csslint') {
+    header = t('issuesHelp', makeLink('https://github.com/CSSLint/csslint/wiki/Rules-by-ID', 'CSSLint'));
+    template = ruleID => {
+      const rule = findCSSLintRule(ruleID);
+      return rule ? `<li><b>${makeLink(rule.url || url, rule.name)}</b><br>${rule.desc}</li>` : '';
+    };
+  } else {
+    header = t('issuesHelp', makeLink(url, 'stylelint'));
+    template = rule => `<li>${makeLink(url + rule, rule)}</li>`;
+  }
+  // to-do: change this to a generator
+  $$('#lint td[role="severity"]').forEach(el => {
+    const rule = el.dataset.rule;
+    if (!rules.includes(rule)) {
+      list += template(rule);
+      rules.push(rule);
+    }
+  });
+  return showHelp(t('issues'), header + list + '</ul>');
+}
+
+function checkLinter(linter = prefs.get('editor.linter')) {
+  linter = linter.toLowerCase();
+  if (prefs.get('editor.linter') !== linter) {
+    prefs.set('editor.linter', linter);
+  }
+  return linter;
+}
+
+function setupLinterSettingsEvents(popup) {
+  $('.save', popup).addEventListener('click', event => {
+    event.preventDefault();
+    const linter = checkLinter(event.target.dataset.linter);
+    const json = tryJSONparse(popup.codebox.getValue());
+    if (json && json.rules) {
+      // it is possible to have stylelint rules popup open & switch to csslint
+      if (linter === 'stylelint') {
+        setStylelintRules(json.rules);
+      } else {
+        setCSSLintRules(json.rules);
+      }
+      updateLinter(linter);
+    } else {
+      $('#help-popup .error').classList.add('show');
+      clearTimeout($('#help-popup .contents').timer);
+      $('#help-popup .contents').timer = setTimeout(() => {
+        // popup may be closed at this point
+        const error = $('#help-popup .error');
+        if (error) {
+          error.classList.remove('show');
+        }
+      }, 3000);
+    }
+  });
+  $('.reset', popup).addEventListener('click', event => {
+    event.preventDefault();
+    const linter = checkLinter(event.target.dataset.linter);
+    let rules;
+    if (linter === 'stylelint') {
+      setStylelintRules();
+      rules = {rules: stylelintDefaultConfig.rules};
+    } else {
+      setCSSLintRules();
+      rules = {rules: csslintDefaultRuleset};
+    }
+    popup.codebox.setValue(JSON.stringify(rules, null, 2));
+    updateLinter(linter);
+  });
+}
+
+function openStylelintSettings() {
+  const linter = prefs.get('editor.linter');
+  BG.chromeLocal.getValue(
+    linter === 'stylelint'
+      ? 'editorStylelintRules'
+      : 'editorCSSLintRules'
+    ).then(rules => {
```

Comment:
causes an eslint error; this single line should be unindented by 2 spaces

Reply:
I've disabled the `indent-legacy` and `no-multi-spaces` eslint rules locally otherwise Atom pops up error messages while I type. I really should fix it or start using VS Code.